bgpd/bgp_nb_config.c

```diff
@@ -186,16 +216,54 @@ int bgp_global_local_as_modify(struct nb_cb_modify_args *args)
 {
 	struct bgp *bgp;
 	as_t as;
+	const struct lyd_node *vrf_dnode;
+	const char *vrf_name;
+	const char *name = NULL;
+	enum bgp_instance_type inst_type;
+	int ret;
+	bool is_view_inst = false;
 
 	switch (args->event) {
 	case NB_EV_VALIDATE:
 		as = yang_dnode_get_uint32(args->dnode, NULL);
-
-		bgp = nb_running_get_entry_non_rec(args->dnode, NULL, false);
+#if 0
```

Comment:
any particular reason we need to leave the #if 0 code in here?  Let's remove the code if it is dead

Reply:
Ack, kept it to debug the api with @rwestphal 
```
  230                 /* NOTE: This returns stale bgp reference.
  231                  * after no router bgp, instance is removed but upon
  232                  * new router bgp stale pointer is returned with junk as value.
  233                  */
  234                 bgp = nb_running_get_entry(args->dnode, NULL, false);
  235                 if (bgp && bgp->as != as) {
  236                         snprintf(args->errmsg, args->errmsg_len,
  237                                  "BGP instance is already running; AS is %u, input-as %u",
  238                                  bgp->as, as);
  239                         return NB_ERR_VALIDATION;
  240                 }
```